src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts

```diff
@@ -0,0 +1,425 @@
+              .propertyKey=${item.property_key}
+              @iron-select=${this._dropdownSelected}
+            >
+              ${Object.entries(item.metadata.states).map(
+                ([key, state]) => html`
+                  <paper-item .value=${key}>${state}</paper-item>
+                `
+              )}
+            </paper-listbox>
+          </paper-dropdown-menu>
+        </div>
+      `;
+    }
+
+    return html`${labelAndDescription}
+      <p>${item.value}</p>`;
+  }
+
+  private _isEnumeratedBool(item): boolean {
+    // Some Z-Wave config values use a states list with two options where index 0 = Disabled and 1 = Enabled
+    // We want those to be considered boolean and show a toggle switch
+    const disabledStates = ["disable", "disabled"];
+    const enabledStates = ["enable", "enabled"];
+
+    if (item.configuration_value_type !== "enumerated") {
+      return false;
+    }
+    if (!("states" in item.metadata)) {
+      return false;
+    }
+    if (!(0 in item.metadata.states) || !(1 in item.metadata.states)) {
+      return false;
+    }
+    if (
+      disabledStates.includes(item.metadata.states[0].toLowerCase()) &&
+      enabledStates.includes(item.metadata.states[1].toLowerCase())
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  private _switchToggled(ev) {
+    this._updateConfigParameter(ev.target, ev.target.checked ? 1 : 0);
+  }
+
+  private _dropdownSelected(ev) {
+    if (ev.target === undefined || this._config![ev.target.key] === undefined) {
+      return;
+    }
+    if (this._config![ev.target.key].value === ev.target.selected) {
+      return;
+    }
+
+    this._updateConfigParameter(ev.target, parseInt(ev.target.selected));
+  }
+
+  private debouncedUpdate = debounce((target) => {
+    const value = parseInt(target.value);
+    this._config![target.key].value = value;
+
+    this._updateConfigParameter(target, value);
+  }, 1000);
+
+  private _numericInputChanged(ev) {
+    if (ev.target === undefined || this._config![ev.target.key] === undefined) {
+      return;
+    }
+    if (this._config![ev.target.key].value === parseInt(ev.target.value)) {
+      return;
+    }
+    this.debouncedUpdate(ev.target);
+  }
+
+  private _updateConfigParameter(target, value) {
+    setNodeConfigParameter(
+      this.hass,
+      this.configEntryId!,
+      this.nodeId!,
+      target.property,
+      value,
+      target.propertyKey ? target.propertyKey : null
+    );
+    this._config![target.key].value = value;
+  }
+
+  private async _fetchData() {
+    if (!this.configEntryId || !this._deviceRegistryEntries) {
+      return;
+    }
+
+    this._device = this._deviceRegistryEntries
+      ? getDevice(this._deviceRegistryEntries, this.deviceId)
+      : undefined;
+
+    if (!this._device) {
+      return;
+    }
+
+    const identifier = getIdentifier(this._device);
+
+    if (!identifier) {
+      return;
+    }
+
+    this.nodeId = parseInt(identifier[1].split("-")[1]);
```

Comment:
Same. No need to keep it as an instance variable. Makes reasoning about this a lot easier.

Reply:
Not sure I follow?